src/WallpaperEngine/Core/CProject.cpp

```diff
@@ -24,7 +24,7 @@ CProject* CProject::fromFile (const irr::io::path& filename)
     CProject* project = new CProject (
         *title,
         *type,
-        CScene::fromFile ((*file).get <std::string> ().c_str ())
+        CScene::fromFile ((*file).get <std::string> ().c_str (), (*type).get <std::string> ().c_str())
```

Comment:
CScene is used because the project type is "scene", in this case a new type should be created for videos (there'll have to be a base class for both CScene and CVideo so this can be stored inside CProject without using void* pointers, in a similar fashion to CObject and CImage, CSound, etc), this way there is no need of having specialized code handling the video inside a scene and you get rid of most of the default initializers for CCamera, CProjection and CScene that shouldn't really be needed.